e2etest/newe2e_task_validation.go

```diff
@@ -191,3 +194,110 @@ func checkMultipleErrors(errorMsg []string, line string) bool {
 
 	return false
 }
+
+func ValidateListTextOutput(a Asserter, stdout AzCopyStdout, expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject, expectedSummary *cmd.AzCopyListSummary) {
+	if dryrunner, ok := a.(DryrunAsserter); ok && dryrunner.Dryrun() {
+		return
+	}
+
+	expectedObj := expectedObjectsToString(expectedObjects)
+	rawObj, rawSum := cleanupRaw(stdout.RawStdout(), expectedSummary != nil)
+
+	// sort to avoid out of order arrays
+	sort.Strings(rawObj)
+	sort.Strings(expectedObj)
+
+	if !reflect.DeepEqual(rawObj, expectedObj) {
+		a.Error(fmt.Sprintf("Does not match - raw:%s. expectedObj:%s.", rawObj, expectedObj))
+	}
+
+	// if expectedObj summary is provided, the last two elements of raw are part of the summary
+	if expectedSummary != nil {
+		validateSummary(a, rawSum, expectedSummary)
+	}
+}
+
+func cleanupRaw(raw []string, hasSummary bool) ([]string, []string) {
+	var objectRaw []string
+	var summaryRaw []string
+	var tempRaw []string
+	for i := 0; i < len(raw); i++ {
+		if raw[i] != "" {
+			tempRaw = append(tempRaw, raw[i])
+		}
+	}
+
+	index := len(tempRaw) - 2 // last two values of raw output should be summary
+	if hasSummary {
+		objectRaw = tempRaw[:index]
+		summaryRaw = tempRaw[index:]
+	} else {
+		objectRaw = tempRaw
+	}
+	return objectRaw, summaryRaw
+}
+
+func validateSummary(a Asserter, rawSummary []string, expectedSummary *cmd.AzCopyListSummary) {
+	expectedSum := [2]string{fmt.Sprintf("File count: %s", expectedSummary.FileCount), fmt.Sprintf("Total file size: %s", expectedSummary.TotalFileSize)}
+
+	if rawSummary[0] != expectedSum[0] {
+		a.Error(fmt.Sprintf("File count does not match - raw:%s. expected:%s.", rawSummary[0], expectedSum[0]))
+	}
+
+	if rawSummary[1] != expectedSum[1] {
+		a.Error(fmt.Sprintf("Total file size does not match - raw:%s. expected:%s.", rawSummary[1], expectedSum[1]))
+	}
+}
+
+func expectedObjectsToString(expectedObjects map[AzCopyOutputKey]cmd.AzCopyListObject) []string {
+	var stringArray []string
+
+	for _, val := range expectedObjects {
+		stringArray = append(stringArray, toString(val))
+	}
+	return stringArray
+}
+
+func toString(lo cmd.AzCopyListObject) string {
```

Comment:
Should we move this logic to the list file to live with the AzCopyListObject?

Reply:
Nitpick: This is also code duplication from AzCopy (`list.go:352-410`).